src/feed/Feed.js

```diff
@@ -32,7 +33,12 @@ class Feed {
     };
 
     onRangeChanged = () => {
-        console.log('RANGE CHANGE', this.startEpoch, this.endEpoch);
+        console.log('RANGE CHANGED', this.startEpoch, this.endEpoch);
+        const dtLeft =  new Date((this.startEpoch || this.endEpoch - ((this.granularity || 1) * 30)) * 1000);
```

Comment:
I hard coded `30` here to just get it to work. Use `this._stx.chart.maxTicks` instead.

There is a piece of code in this file also using `(granularity || 1) * this._stx.chart.maxTicks`. See if you reuse that.